methods.rb

```diff
@@ -1 +1,31 @@
-# Your code here!
+def greet_programmer
+  puts "Hello, programmer!"
+end
+
+
+def greet(name)
+  puts "Hello, #{name}!"
+end
+
+
+def greet_with_default(name = nil)
+  if name
+    puts "Hello, #{name}!"
+  else
+    puts "Hello, programmer!"
+  end
+end
+
+
+def add(num1, num2)
+  return num1 + num2
+end
```

Comment:
The final line of a ruby method is always returned. No need to explicitly return here. 
```suggestion
def add(num1, num2)
  num1 + num2
end
```